modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockService.scala

```diff
@@ -81,13 +80,25 @@ object BlockService {
 
         addressBalanceChanges.toList.traverse {
           case (address, balanceChange) =>
-            for {
-              current <- addressStorage.getBalance(address)
-              newBalance <- refineV[NonNegative]
-                .apply(current.coerce + balanceChange)
-                .bimap(BalanceNegative, Balance(_))
+            val positive = balanceChange > 0
```

Comment:
I'm not sure if this approach with figuring out if the change is positive or negative, then constructing the `NonNegLong` and then wrapping it in `Balance` just to add or subtract is the way to go. I would rather base everything on simple types and have one point of conversion -> if after all the adding it's still correct lets get Some(Balance) otherwise None.

Reply:
We could even go with `BigInt` for adding subtracting and so one, and pass the value to a function `Balance.fromBigInt()` and have this single function decide whether that is still a correct balance or not.

Reply:
We agreed on Slack to keep a single type and avoid conversions